sympy/solvers/ode/single.py

```diff
@@ -825,23 +827,29 @@ class SecondAutonomousNth(SinglePatternODESolver):
 
     >>> from sympy import exp, log, cbrt
     >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
-    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    >>> dsolve(eq, hint='2nd_autonomous_nth')
     [Eq(Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 + x),
-    Eq(Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 - x)]
+    Eq(-Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 + x)]
 
     >>> eq = f(x).diff(x, 2) + cbrt(f(x)) + 1/f(x)
-    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    >>> dsolve(eq, hint='2nd_autonomous_nth')
     [Eq(sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 + x),
-    Eq(sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 - x)]
+    Eq(-sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 + x)]
 
     >>> eq = f(x).diff(x, x) + f(x).diff(x)**4
-    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
-    [Eq((C1 + 2*f(x))**(3/2)/3, C2 + x), Eq((C1 + 2*f(x))**(3/2)/3, C2 - x)]
+    >>> dsolve(eq, hint='2nd_autonomous_nth')
+    Eq((C1 + 2*f(x))**(S(3)/2)/3, C2 + x)
+
+    >>> eq = f(x).diff(x, x) + f(x).diff(x)**n
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify = False)
```

Comment:
This can also be solved by not setting `simplify` to `False`.
```
>>> eq = f(x).diff(x, x) + f(x).diff(x)**n
>>> dsolve(eq, hint='2nd_autonomous_nth')
[Eq(f(x), Piecewise((C1/(n - 2) + (C2*n - C2 + n*x - x)**(n/(n - 1))/(n*(C2*n - C2 + n*x - x)**(2/(n - 1)) - 2*(C2*n - C2 + n*x - x)**(2/(n - 1))), Ne(1/(n - 2), -1)), (nan, True))), 
Eq(f(x), Piecewise((C1/(n - 2) + exp(C2*n + n*x)/(n*exp(2*C2 + 2*x) - 2*exp(2*C2 + 2*x)), Eq(1/(n - 2), -1)), (nan, True)))]
```

Reply:
These are not correct for `n=1`.